### ISSUE TITLE
Method overview in documentation

### ISSUE THREAD

**Issue by lbalicki** [pymor/pymor#1456]
We should add a list of all methods of a class in the documentation.

**Comment by renefritze** [comment]
Just to make sure we're all on the same page here. 
This means a table like 'Methods' in the `BlockDiagonalOperator` section 
[here](https://docs.pymor.org/2020.2.0/generated/pymor.operators.html#pymor.operators.block.BlockDiagonalOperator). 
Right?
In that old example this listed inherited methods too.

**Comment by sdrave** [comment]
Yes.

**Comment by renefritze** [comment]
If somebody wants to tackle this, they should first look into whether this case can be resolved simply by changing `autoapi` [configuration](https://sphinx-autoapi.readthedocs.io/en/latest/reference/config.html#confval-autoapi_options) in our `docs/source/conf.py`. 
If that is not the case, they can try to override the [responsible template](https://raw.githubusercontent.com/readthedocs/sphinx-autoapi/master/autoapi/templates/python/class.rst) with a modified version, by copying  that `class.rst` into `docs/source/_templates/autoapi/python/` and then adding output.

**Comment by MagnusOstertag** [comment]
Hi @renefritze,
I'd like to contribute sth to pyMOR. And figured this would be a good first issue for me. Did you already make some progress and I should tackle another issue? Or can I work on this issue?

**Comment by renefritze** [comment]
Hi @MagnusOstertag! This is indeed a good first issue, as it requires basically no knowledge of the inner workings of pyMOR. I have not gone further on this than above ideas and it would be awesome if you want to try your hand at it. Let me know if you need any more pointers!

**Comment by MagnusOstertag** [comment]
I added the method table by tweaking the `class.rst` file. 
I kept the style consistent with the overviews of `Classes` and `Functions`.

However, I was not able to emphasize the `Methods` as a proper (sub-)section title.
When I add a 7-bit ASCII character recommended (as described in the `.rst` documentation), I get a `CRITICAL: Unexpected section title.` error.
Could it be that the indentation as taken from [`class.rst`](https://raw.githubusercontent.com/readthedocs/sphinx-autoapi/master/autoapi/templates/python/class.rst) is somehow strange and causes the error in the modified version?
